src/ion_data/mod.rs

```diff
@@ -0,0 +1,110 @@
+mod ion_eq;
+mod ion_ord;
+
+use crate::{Element, IonResult};
+use std::cmp::Ordering;
+use std::fmt::{Display, Formatter};
+
+use crate::element::{Annotations, Value};
+use delegate::delegate;
+pub use ion_eq::IonEq;
+pub(crate) use ion_ord::IonOrd;
+
+/// A wrapper for lifting "data" to "Ion data". Anything that implements [IonEq] can be converted
+/// into [IonData]. As [IonData], the semantics of [Eq] are the same as [IonEq]—that is to say, [Eq]
+/// follows the rules for Ion Equivalence in the Ion specification.
+///
+/// [IonData] itself does not implement [IonEq] so that it is not possible to create, for example,
+/// `IonData<IonData<Element>>`.
+///
+/// [Hash] and [Ord] are not guaranteed to be implemented for all [IonData], but when they are, they
+/// are required to be consistent with [IonEq] (and [Eq]).
+///
+/// __WARNING__—The Ion specification does _not_ define a total ordering over all Ion values. Do
+/// not depend on getting any particular result from [Ord]. Use it only as an opaque total ordering
+/// over all [IonData]. _The algorithm used for [Ord] may change between versions._
```

Comment:
I am persnickety about liking doc comments around types (especially linked ones) to be in typewriter font: 

```suggestion
/// A wrapper for lifting "data" to "Ion data". Anything that implements [`IonEq`] can be converted
/// into [`IonData`]. As [`IonData`], the semantics of [`Eq`] are the same as [`IonEq`]—that is to say, [`Eq`]
/// follows the rules for Ion Equivalence in the Ion specification.
///
/// [`IonData`] itself does not implement [`IonEq`] so that it is not possible to create, for example,
/// `IonData<IonData<Element>>`.
///
/// [Hash] and [Ord] are not guaranteed to be implemented for all [`IonData`], but when they are, they
/// are required to be consistent with [`IonEq`] (and [`Eq`]).
///
/// __WARNING__—The Ion specification does _not_ define a total ordering over all Ion values. Do
/// not depend on getting any particular result from [`Ord`]. Use it only as an opaque total ordering
/// over all [`IonData`]. _The algorithm used for [`Ord`] may change between versions._
```

Reply:
Will do!